### PR TITLE
kernel: de-duplicate code reading `local` statements; make reader functions static

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: src/read.c [hunk 1]
```diff
@@ -1499,6 +1499,41 @@ static void ReadFuncExprBody(
 }
 
 
+/****************************************************************************
+**
```
Comment (ChrisJefferson) [inline src/read.c]: Being super picky, maybe mention nams already contains the list if functiin arguments, and either mention narg==LEN_LIST(nams), or drop it and recalculate it.
?

Comment (fingolfin) [reply src/read.c]: Actually, I didn't think about this at all. Good catch! I'll drop the `narg` argument.

Comment (fingolfin) [reply src/read.c]: Done.

**Comment by fingolfin** [comment]
The test failure is rather strange:
```
########> Diff in /home/travis/build/gap-system/gap/tst/testbugfix/2012-07-13-t00248.tst:2
# Input is:
IsDocumentedWord( "d" );
# Expected output:
false
# But found:
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `SplitString' on 3 arguments
```
So far, I failed to reproduce this locally, on macOS and Ubuntu. My only guess at this point is that somehow this PR relies on undefined behavior, but I can't figure out how. If @ChrisJefferson or @stevelinton or anybody else has an idea...

**Pull request merged**